src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/ListViewModel.cs

```diff
@@ -367,6 +374,43 @@ private void SetSelectedItem(ListItemViewModel item)
 
                TextToSuggest = item.TextToSuggest;
            });
+
+        _lastSelectedItem = item;
+        _lastSelectedItem.PropertyChanged += SelectedItemPropertyChanged;
+    }
+
+    private void SelectedItemPropertyChanged(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
+    {
+        var item = _lastSelectedItem;
+        if (item == null)
+        {
+            return;
+        }
+
+        // already on the UI thread here
+        switch (e.PropertyName)
+        {
+            case nameof(item.Command):
+            case nameof(item.SecondaryCommand):
+            case nameof(item.AllCommands):
+            case nameof(item.Name):
+                WeakReferenceMessenger.Default.Send<UpdateCommandBarMessage>(new(item));
+                break;
+            case nameof(item.Details):
+                if (ShowDetails && item.HasDetails)
+                {
+                    WeakReferenceMessenger.Default.Send<ShowDetailsMessage>(new(item.Details));
+                }
+                else
+                {
+                    WeakReferenceMessenger.Default.Send<HideDetailsMessage>();
+                }
+
+                break;
+            case nameof(item.TextToSuggest):
+                TextToSuggest = item.TextToSuggest;
+                break;
```

Comment:
do you want a default case?

Reply:
nah, these are the only properties that the List Page VM cares about on the selected item. Everything else doesn't bubble up through the page